src/itensors.jl

```diff
@@ -41,3 +41,20 @@ trivial_space(x::Index) = _trivial_space(x)
 trivial_space(x::Vector{<:Index}) = _trivial_space(x)
 trivial_space(x::ITensor) = trivial_space(inds(x))
 trivial_space(x::Tuple{Vararg{Index}}) = trivial_space(first(x))
+
+"""
+Given an input tensor and a Dict (ind_to_newind), replace inds of tensor that are also
+keys of ind_to_newind to the value of ind_to_newind.
+Note that it is the same as
+ITensors.replaceinds(tensor, collect(keys(ind_to_newind)) => collect(values(ind_to_newind))).
+Based on benchmark, this implementation is more efficient when the size of ind_to_newind is large.
+TODO: we can remove this function once the original replaceinds performance is improved.
+"""
```

Comment:
I realized another problem with the current `replaceinds` implementation is that `collect(keys(ind_to_newind))` cannot be too large: https://github.com/ITensor/ITensors.jl/blob/8535b7fd45915bcd628b5544b84f3ff6189011f9/NDTensors/src/tupletools.jl#L4

Reply:
Thanks, yeah we should fix that in ITensors.jl, I've seen this pop up other places as well. Using `intersect`/`commoninds` like you did here would be a simple fix.